lib/FusionInventory/Agent/Task/Inventory/Virtualization/Xen.pm

```diff
@@ -8,7 +8,7 @@ use FusionInventory::Agent::Tools;
 our $runMeIfTheseChecksFailed = ["FusionInventory::Agent::Task::Inventory::Virtualization::Libvirt"];
 
 sub isEnabled {
-    return canRun('xm');
+    return canRunOK('xm') || canRunOK('xl');
 }
```

Comment:
David is correct here. With the current code, version or correctness doesn't matter for the first phase (isEnabled() function), as the simple presence of either 'xm' or 'xl' command is enough to ensure the second phase (doInventory() function) is executed. Unfortunately, this lead to execute  the same test again in the second phase, to identify the working command.

A better solution would be to identify the working command in the first function, and to memorise it for the second, ie:

my $command;

 sub isEnabled {
    $command = canRunOK('xm') ? 'xm'   :
                          canRunOK('xl')  ? 'xl'      :
                                                       undef;
    return defined($command); 
 }